src/sonic-bgpcfgd/bgpcfgd/managers_device_global.py

```diff
@@ -0,0 +1,106 @@
+from .manager import Manager
+from .log import log_err, log_debug, log_notice
+import re
+
+class DeviceGlobalCfgMgr(Manager):
+    """This class responds to change in device-specific state"""
+
+    def __init__(self, common_objs, db, table):
+        """
+        Initialize the object
+        :param common_objs: common object dictionary
+        :param db: name of the db
+        :param table: name of the table in the db
+        """
+        self.directory = common_objs['directory']
+        self.cfg_mgr = common_objs['cfg_mgr']
+        self.constants = common_objs['constants']
+        self.tsa_template = common_objs['tf'].from_file("bgpd/tsa/bgpd.tsa.isolate.conf.j2")
+        self.tsb_template = common_objs['tf'].from_file("bgpd/tsa/bgpd.tsa.unisolate.conf.j2")
+        super(DeviceGlobalCfgMgr, self).__init__(
+            common_objs,
+            [],
+            db,
+            table,
+        )
+
+    def set_handler(self, key, data):
+        log_debug("DeviceGlobalCfgMgr:: set handler")
+        """ Handle device tsa_enabled state change """
+        if not data:
+            log_err("DeviceGlobalCfgMgr:: data is None")
+            return False
+
+        if "tsa_enabled" in data:
+            self.cfg_mgr.commit()
```

Comment:
What's the purpose of commit/update here?

Reply:
This is mainly for the reboot sequence where tsa_enabled is set after all peer-group configuration, but before it is actually committed to configDB. Commit is used here to ensure TSA routemap generation is done with all the pg config